### DISCUSSION TITLE
Generating local notifications with RN Firebase

### DISCUSSION THREAD

**Discussion by bilaljawed** [invertase/react-native-firebase#5472] [Q&A]
Hello,

First of all thank you for the fantastic plugin and the documentation, my question is that:  i am sending an encrypted notification through fcm to user which can only be decrypted by users device, i have sent only data property through fcm notification payload because if i include "notification" parameter, the notification pops up automatically, so what i want to do is receive message in the background, decrypt it and generate a local notification with the decrypted message,

I have read that I can use this handler to intercept background message **setBackgroundMessageHandler**

but how do i generate a local notification

Thank you!

**Comment by mikehardy** [comment]
@bilaljawed you must integrate a full-featured notification package like https://notifee.app or one of the free alternatives like react-native-push-notification

Please be aware: iOS **does not guarantee delivery of data-only FCM**. I won't cover all the related information - if you're doing an encrypted messaging app you are likely capable of searching the issues list for "data-only" and "reliable" and you'll quickly find all the info

Good luck!
